compiler/rustc_const_eval/src/interpret/terminator.rs

```diff
@@ -174,8 +166,8 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 terminator.kind
             ),
 
-            InlineAsm { template, ref operands, options, destination, .. } => {
-                M::eval_inline_asm(self, template, operands, options)?;
+            InlineAsm { template, operands, options, destination, .. } => {
```

Comment:
There are now double references here where there weren't ones previously.

I don't really understand the point of these changes, this `match` expression in particular. Having to dereference values of types that are `Copy` seems like adding complexity to me, not removing it. To me, a lot of these changes seem like very slight improvements at best. :)

Reply:
Yeah I agree, it's better if the pattern can avoid creating `&Copy` references. Unfortunately due to https://github.com/rust-lang/rust/issues/64586 that often requires `ref` keywords, since match ergonimcs does not work well with `&` patterns. Still I think this is better than having `&Copy` references.